### DISCUSSION TITLE
Multisite / multi website support

### DISCUSSION THREAD

**Discussion by OZZlE** [vercel/next.js#34124] [Ideas]
### Describe the feature you'd like to request

I would like to have more out of the box multi website support. For example if you have several websites that share a lot of similar  functionality, e.g.:

localhost:3000/website/ => `www.website.com`
localhost:3000/other/ => `www.otherwebsite.com`

### Describe the solution you'd like

Not entirely sure, to be able to configure this simply by config files and perhaps folder structure in /pages/ ?

I guess any solution would be good but as few steps and simple as possible :) Would need to be able to do:

- Localisation per website - e.g. `localhost:3000/website1/[lang]/`
- Be able to have different global styles per website, preferably one super global stylesheet for all and then support for one for each website.  

I am still investigating a bit what is possible today and there is probably more things that I've missed.

### Describe alternatives you've considered

You can sort of already do this with some semi-major downsides:

- You need to implement localisation manually because the standard solution becomes `localhost:3000/en/website1/` instead of `localhost:3000/website1/en/` and then you cannot have all the languages of that website live on the same domain unless you use subdomains which you probably don't want either.
- Localisation per website, one website might only have 1 language while another has 3
- Global styles affect all websites, there is no built in possibility of adding global styles to only one website (without it having to be downloaded at least for the other websites)

**Comment by paulpopus** [comment]
At this point isn't it easier to setup two different sites on your subroutes? They can still share the code base

With env vars you can customise how/what is configured per site and you can have your localisations and all working per site.

**Comment by OZZlE** [comment]
Not entirely sure what you mean exactly @paulpopus , I have done this for now:

/pages/website1/index.tsx
/pages/website2/index.tsx

but if I add to next.config

    i18n: {
      defaultLocale: "en-en",
      locales: ["en-en", "no-no", "sv-se", "da-dk", "fi-fi"],
    },

I can visit `localhost/sv-se/website1/` but not `localhost/website1/sv-se/` and a prod or stage will have the root as `localhost/website1/` => `stg.website.com`  and then you want to add the current language to the url `stg.website.com/sv-se/` but that would map to `localhost/website1/sv-se/` which doesn't work with default i18n

only option I've found so far is skipping i18n configuration and manually adding

/pages/website1/[...language].tsx and then adding a custom _document to set the proper `<html lang=x>` etc etc

**Comment by balazsorban44** [comment]
Have you seen https://nextjs.org/docs/advanced-features/multi-zones? Would that solve your issue?

**Reply by OZZlE** [reply]
Thanks I will check it out!

**Reply by OZZlE** [reply]
Question is, since zones has their own package.json, does it mean users will have to download and parse react lib js multiple times? or does it just add differences?

**Reply by OZZlE** [reply]
I'm considering using it instead for Micro Frontends that's why I'm wondering

**Comment by paulpopus** [comment]
^ 

But also, I mean that you could deploy two separate sites even with the same codebase (or very similar), different .envs by changing your base path https://nextjs.org/docs/api-reference/next.config.js/basepath

So you can deploy one website to /website1 and another to /website2 on the same domain. Another way to do this is by proxying
